### ISSUE TITLE
Avoid node query with Namespaced permissions

### ISSUE THREAD

**Issue by johnSchnake** [vmware-tanzu/sonobuoy#1552]
See #1470 for background.

When running sonobuoy with --aggregator-permissions=namespaced we should be able to as much as possible with ONLY namespaced permissions. This means no cluster permissions at all, even to query nodes.

This can be quite limiting but we can/should support this use case in some capacity. Even if it means disabling daemonsets or monitoring them differently.

**Comment by johnSchnake** [comment]
Confirmed with initiating user that there is a usecase for it even if very limiting for daemonsets or e2e.

I think that it makes sense to bifurcate the feature into two:
 - a strict namespaced-only version which limits daemonsets, queries, and probably wont work with e2e
 - a less strict namespaced version which allows querying some cluster resources. Potentially just nodes to ensure sonobuoy can run, run daemonsets, and start e2e tests (even if they dont necessarily pass. Potentially e2e upstream can/does support a case where you dont create namespaces)

**Comment by johnSchnake** [comment]
I applied p1 because I want to get this fixed asap before the next release.

Also, I realize that in the latest changes I removed the creation of the namespace from the manifest. This _may_ not have been strictly necessary since its not the aggregator that does that namespace creation, but the local permissions. I think it is in the spirit of the limited permissions that we avoid that namespace creation though in case even the local permissions are that restricted.
